### DISCUSSION TITLE
Pull parser in Rust

### DISCUSSION THREAD

**Discussion by kmaasrud** [jgm/djot#150] [Show and tell]
Hey, just popping in here to show off a new Djot project I've started. It's a pull parser for Djot, written in Rust.

I have no intentions of challenging @matklad's [djot-rs](https://github.com/matklad/djot-rs), as this is quite another approach---more focused on pure parsing speed than compatibility with the source code. It is aimed to be more akin to [pulldown-cmark](https://github.com/raphlinus/pulldown-cmark), and will not produce an AST.

The library source is hosted on [Sourcehut](https://git.sr.ht/~kmaasrud/djot.rs), and I've started a CLI hosted on [GitHub](https://github.com/kmaasrud/djot.rs).

I'm posting this here in the hopes that someone might have some feedback or contributions. If so, please reach me at the [mailing list](https://lists.sr.ht/~kmaasrud/djot-discuss) or below in this discussion. This is a challenge I've never tackled before, so it might not pan out to anything at all, but it is so far a really fun side project!

**Comment by matklad** [comment]
Linked that from djot-rs readme! I've run out of steam for djot for the time being, so I'd be more than happy if djr makes djot-rs unnecessary :-)

**Comment by jgm** [comment]
Exciting!  Keep us posted.
